monai/transforms/smooth_field/dictionary.py

```diff
@@ -81,67 +99,76 @@ def randomize(self, data: Optional[Any] = None) -> None:
         if self._do_transform:
             self.trans.randomize()
 
-    def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Mapping[Hashable, np.ndarray]:
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Mapping[Hashable, NdarrayOrTensor]:
         self.randomize()
 
         if not self._do_transform:
             return data
 
         d = dict(data)
 
-        for key in self.key_iterator(d):
+        for idx, key in enumerate(self.key_iterator(d)):
             if not self.apply_same_field:
                 self.randomize()  # new field for every key
```

Comment:
for this `apply_same_field=False`, is it an equivalent of creating multiple instances
```py
rand_smooth = RandSmoothFieldAdjustContrastd("img", ...)
rand_smooth1 = RandSmoothFieldAdjustContrastd("img1", ...)
```
I think we should drop this option to simplify the API...

Reply:
this comment is not resolved before merging... could you help clarify the use case of `apply_same_field`? @ericspod

Reply:
It was an edge case to consider if there was some reason to distort two images in different ways, eg. for registration training, and it would be more efficient than having two transforms. I can remove it in a separate PR if you think it's worth removing.

Reply:
https://github.com/Project-MONAI/MONAI/pull/3556